src/compiler/factory.ts

```diff
@@ -3581,11 +3581,17 @@ namespace ts {
         }
     }
 
+    /**
+     * accessor declaration that can be converted to an expression (`name` field cannot be a `PrivateName`)
+     */
+    type ExpressionableAccessorDeclaration = AccessorDeclaration & {name: Exclude<PropertyName, PrivateName>};

```

Comment:
Might be more consistent to define like this:

```typescript
interface ExpressionableAccessorDeclaration extends AccessorDeclaration {
    name: Exclude<PropertyName, PrivateName>;
}
```

Reply:
Yours looks better to me, too, but unfortunately AccessorDeclaration is a type, not an interface, and so cannot be extended. So I'd end up with a type intersection anyway.